modin/engines/ray/pandas_on_ray/frame/partition.py

```diff
@@ -239,30 +252,57 @@ def get_index_and_columns(df):
 
 
 @ray.remote(num_returns=4)
-def deploy_ray_func(call_queue, partition):  # pragma: no cover
+def apply_func(partition, func, *args, **kwargs):  # pragma: no cover
+    try:
+        result = func(partition, *args, **kwargs)
+    # Sometimes Arrow forces us to make a copy of an object before we operate on it. We
+    # don't want the error to propagate to the user, and we want to avoid copying unless
+    # we absolutely have to.
+    except ValueError:
+        result = func(partition.copy(), *args, **kwargs)
+    return (
+        result,
+        len(result) if hasattr(result, "__len__") else 0,
+        len(result.columns) if hasattr(result, "columns") else 0,
+        get_node_ip_address(),
+    )
+
+
+@ray.remote(num_returns=4)
+def apply_list_of_funcs(partition, funcs):  # pragma: no cover
     def deserialize(obj):
         if isinstance(obj, ObjectIDType):
             return ray.get(obj)
-        return obj
+        elif isinstance(obj, tuple) and any(isinstance(o, ObjectIDType) for o in obj):
+            return ray.get(list(obj))
+        elif isinstance(obj, dict) and any(
+            isinstance(obj[o], ObjectIDType) for o in obj
+        ):
+            return ray.get(list(obj.values()))
```

Comment:
```suggestion
            return dict(zip(obj.keys(), ray.get(list(obj.values()))))
```
not sure if this is needed... but I think it's better to retain the object type (`dict` in this case) upon deserialization, otherwise e.g. deserializing `**kwargs` would break

Reply:
I agree, will be changed.

Reply:
Fixed.